### PR TITLE
Add a helper function to get output from shell

### PR DESCRIPTION
**Pull request by NickSeagull**
This would allow to write things like

```haskell
main = do
    Just directory <- getShellOut "stack path --local-install-root"
    print directory
```

instead of 

```haskell
main = do
    Just directory <- fold (inshell "stack path --local-install-root" empty) Foldl.head
    print directory
```

I hope this helps in some way :smile:

### CODE REVIEW

**Comment by Gabriel439** [comment]
I think this utility probably belongs in the `Turtle.Prelude` module, but I might propose restructuring it like this:

```haskell
single :: MonadIO a => Shell a -> io (Maybe a)
single s = do
    ls <- fold s Foldl.list
    case ls of
        [a] -> return (Just a)
        _   -> return Nothing
```

That way you can use it with any function that returns a `Shell` and it also fails if there is more than one line of output.

**Comment by NickSeagull** [comment]
@Gabriel439 I just pushed the changes that you requested 😃

File: src/Turtle/Prelude.hs [hunk 1]
```diff
@@ -1825,3 +1825,10 @@ header (Shell k) = Shell k'
             return (Pair x Nothing)
 
         done' (Pair x _) = done x
+
+single :: MonadIO a => Shell a -> io (Maybe a)
+single s = do
+    ls <- fold s Contorl.Foldl.list
```
Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: `s/Contorl/Control/`

**Comment by NickSeagull** [comment]
I agree with those changes @Gabriel439 , it is true that the script should fail explicitly, because otherwise it is some unexpected behaviour. Just pushed those changes

File: src/Turtle/Prelude.hs [hunk 2]
```diff
@@ -1825,3 +1825,12 @@ header (Shell k) = Shell k'
             return (Pair x Nothing)
 
         done' (Pair x _) = done x
+
+single :: MonadIO io => Shell a -> io (Maybe a)
```
Comment (Gabriel439) [inline src/Turtle/Prelude.hs]: One last request: could you add documentation for this utility?

**Comment by NickSeagull** [comment]
There we go! 😄

**Comment by Gabriel439** [comment]
Thank you! :)

**Comment by mitchellwrosen** [comment]
Just wondering, why does this return a `Maybe`?

**Comment by Gabriel439** [comment]
Oops!  This should probably just return an `a` instead of a `Maybe a`

**Comment by Gabriel439** [comment]
Type of `single` fixed in 71db10ab74b4f9aea75c2b116cca3c392d60c612

**Pull request merged**